src/test/java/de/adorsys/keycloak/config/provider/KeycloakProviderIT.java

```diff
@@ -22,20 +22,62 @@
 
 import de.adorsys.keycloak.config.AbstractImportTest;
 import de.adorsys.keycloak.config.exception.KeycloakProviderException;
+import org.apache.http.conn.ConnectTimeoutException;
+import org.checkerframework.checker.units.qual.Time;
 import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
 import org.junitpioneer.jupiter.SetSystemProperty;
+import org.keycloak.admin.client.resource.RealmsResource;
+import org.keycloak.representations.idm.RealmRepresentation;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.test.context.TestPropertySource;
 
 import javax.ws.rs.NotFoundException;
 import javax.ws.rs.ProcessingException;
 
+import java.net.SocketTimeoutException;
+import java.util.List;
+
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.matchesPattern;
-import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.*;
 
 class KeycloakProviderIT {
+    @Nested
+    @TestPropertySource(properties = {
+            "keycloak.read-timeout=PT0.01S"
+    })
+    class ResteasyReadTimeout extends AbstractImportTest {
+        @Test
+        void run() {
+            // very low read timeout leads inevitably to a read timeout, which in turn shows that the configuration is applied
+            ProcessingException thrown = assertThrows(ProcessingException.class, keycloakProvider::getInstance);
+            assertNotNull(thrown.getCause());
+            assertTrue(thrown.getCause() instanceof SocketTimeoutException);
+            assertThat(thrown.getCause().getMessage(), containsString("Read timed out"));
+        }
+    }
+
+    @Nested
+    @TestPropertySource(properties = {
+            "keycloak.url=https://10.255.255.1",
+            "keycloak.connect-timeout=PT0.01S"
+    })
+    class ResteasyConnectTimeout extends AbstractImportTest {
+        @Test
+        @Timeout(value = 1L)
+        void run() {
+            // connect timeout since IP is not reachable - test fails if it exceeds one second which in turn shows that
+            // the configuration is applied
+            ProcessingException thrown = assertThrows(ProcessingException.class, keycloakProvider::getInstance);
+            assertNotNull(thrown.getCause());
+            assertTrue(thrown.getCause() instanceof ConnectTimeoutException);
+            assertThat(thrown.getCause().getMessage(), containsString("Connect timed out"));
```

Comment:
Looks like java 11 and java 17 have different messages. I guess a case insensitive check would be the best here. Or just remove the C.

Reply:
weird :-) fixed